src/skywallet/transaction.go

```diff
@@ -0,0 +1,175 @@
+	Version    int
+	LockTime   int
+	signatures []string
+	state      int
+}
+
+// Sign method signs the Skycoin Transaction
+func (s *SkycoinTransactionSigner) Sign() ([]string, error) {
+	msg, err := s.initSigningProcess()
+
+	if err != nil {
+		return nil, err
+	}
+
+	index := 0
+	s.state = 0
+
+	for {
+		switch msg.Kind {
+		case uint16(messages.MessageType_MessageType_TxRequest):
+			txRequest := &messages.TxRequest{}
+			err = proto.Unmarshal(msg.Data, txRequest)
+			if err != nil {
+				return nil, err
+			}
+			switch *txRequest.RequestType {
+			case messages.TxRequest_TXINPUT:
+				if s.state == 0 { // Sending Inputs for InnerHash
+					if len(s.Inputs)-index > 8 {
+						msg, err = s.sendInputs(index, 8)
+						if err != nil {
+							return nil, err
+						}
+						index += 8
+					} else {
+						msg, err = s.sendInputs(index, len(s.Inputs)-index)
+						if err != nil {
+							return nil, err
+						}
+						s.state++
+						index = 0
+					}
+				} else if s.state == 2 { // Sending Inputs for Signatures
+					err = s.addSignatures(&msg)
+					if err != nil {
+						return nil, err
+					}
+					if len(s.Inputs)-index > 8 {
+						msg, err = s.sendInputs(index, 8)
+						if err != nil {
+							return nil, err
+						}
+					} else {
+						msg, err = s.sendInputs(index, len(s.Inputs)-index)
+						if err != nil {
+							return nil, err
+						}
+						s.state++
+						index = 0
+					}
+					index += 8
+				} else {
+					return nil, fmt.Errorf("protocol error: unexpected TxRequest type")
+				}
+			case messages.TxRequest_TXOUTPUT:
+				if s.state == 1 { // Sending Outputs for InnerHash
+					if len(s.Outputs)-index > 8 {
+						msg, err = s.sendOutputs(index, 8)
+						if err != nil {
+							return nil, err
+						}
+						index += 8
+					} else {
+						msg, err = s.sendOutputs(index, len(s.Outputs)-index)
+						if err != nil {
+							return nil, err
+						}
+						s.state++
+						index = 0
+					}
+				} else {
+					return nil, fmt.Errorf("protocol error: unexpected TxRequest type")
+				}
+			case messages.TxRequest_TXFINISHED:
+				if s.state == 3 {
+					err = s.addSignatures(&msg)
+					if err != nil {
+						return nil, err
+					}
+					return s.signatures, nil
+				}
+				return nil, fmt.Errorf("protocol error: unexpected TXFINISHED message")
+			}
+		case uint16(messages.MessageType_MessageType_Failure):
+			failMsg, err := DecodeFailMsg(msg)
+			if err != nil {
+				return nil, err
+			}
+			return nil, fmt.Errorf("failed with message: %s", failMsg)
+		case uint16(messages.MessageType_MessageType_ButtonRequest):
+			msg, err = s.Device.ButtonAck()
+			if err != nil {
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unexpected response message type from hardware wallet")
+		}
+	}
+}
+
+func (s *SkycoinTransactionSigner) initSigningProcess() (wire.Message, error) {
+	return s.Device.SignTx(len(s.Outputs), len(s.Inputs), "Skycoin", s.Version, s.LockTime, "dkdji9e2oidhash")
```

Comment:
what is this hardcoded value here?

Reply:
For long transaction support, the SignTx message was added, which initiate transaction signing with tx_hash, which is not used. As a function to make the message needed some tx_hash, some random tx was put. 

Maybe, a good idea would be just to pass empty string

Reply:
ah okay.
would be nice to put it in the comments somewhere